AmplifyTools/AmplifyXcode/Sources/AmplifyXcode/Support/ArgumentParser+CLICommand.swift

```diff
@@ -11,10 +11,11 @@ import ArgumentParser
 /// The following extensions on ArgumentParser commands parameters property wrappers
 /// help us providing a hook to generate a JSON representation of a CLI command.
 /// As for now `@PropertyWrappers` APIs to access enclosing type are still "private", so the passed
-/// `registry` allows the property to reference an external type.
+/// `parameters` allows the property to reference an external type.
 /// `Argument` has been left out on purpose as we'd rather use options and flags for clarity of use.
+/// Also by providing these extra initializers we make parameter name explicit.
 extension Option where Value: ExpressibleByArgument {
-    init(wrappedValue: Value, name: String, help: String, _ registry: CLICommandEncodableParametersRegistry) {
+    init(wrappedValue: Value, name: String, help: String, _ parameters: inout Set<CLICommandParameter>) {
```

Comment:
Minor naming suggestion to clarify the intent of the `parameters` and why it's an `inout`:
```suggestion
    init(wrappedValue: Value, name: String, help: String, updating parameters: inout Set<CLICommandParameter>) {
```

Result at the call site would look like:
```swift
@Option(name: "path", help: "Project base path", updating: &parameters)
...
```

Reply:
good call, thanks!